components/Valuation/DataComparison/DataComparisonCard.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { IAPIData, IPredictions } from "../../../lib/types";
+
+interface Props {
+  apiData: IAPIData | undefined;
+  predictions: IPredictions | undefined;
+}
+
+function getMetaverseTokenFromOpenseaLink(opensea_link: string | undefined) {
+  let metaverse = "";
+  let tokenID = "";
+  if (opensea_link) {
+    const split = opensea_link.split("/");
+    metaverse = split[4];
+    tokenID = split[5];
+  }
+  return { metaverse, tokenID };
+}
+
+/**
+ * 
+ * @param param0 
+ * @returns 
+ */
+const DataComparisonCard = ({ apiData, predictions }: Props) => {
+  const [offers, setOffers] = useState<number>();
+  const [openseaLink, setOpenSeaLink] = useState<string | undefined>(apiData?.external_link);
+  
+  const usdPredictionPrice = predictions?.usdPrediction;
+  const handleData = async () => {
+    let { metaverse, tokenID } = getMetaverseTokenFromOpenseaLink(
+      apiData?.opensea_link
+    );
+    const res = await fetch("/api/getLandOffers", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ metaverse: metaverse, tokenId: tokenID }),
+    });
+    const data = await res.json();
+    return data;
+  };
+
+  let usdPrice;
+  let offerPrice;
+
+  useEffect(() => {
+    const data = handleData();
+    data
+      .then((res) => {
+        offerPrice = res.offers[0].current_price;
+        offerPrice = parseFloat(offerPrice) / 1000000000000000000;
```

Comment:
This isn't critical here since we are just displaying data and not really inputting, but its good to keep in mind. Using too many zeros can be risky since its very easy to loose count of them or read them if there's no separators. That's why usually you would see `100_000_000` or `100,000,000`. In crypto since we are gonna be using usually 18 Zeros/Decimals a lot its much clearer for other devs and also much safer for everyone to write it as  `1e18` or `10**18`

We also use ethers.js that has methods like `parseEther` and `formatEther` that do similar things. You can check them out as well.